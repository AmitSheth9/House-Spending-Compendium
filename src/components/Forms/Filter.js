import React from 'react'
import './forms.css'

export default function Filter({setSelectedSort, selectedSort}) {
    return (
        <div>
            <select
                value={selectedSort}
                onChange={e => setSelectedSort(e.target.value)}>
                    <option value='disbursements'>Disbursements</option>
                    <option value='receipts'>Receipts</option>
                    <option value='cash_on_hand_end_period'>Cash on Hand</option>
                    <option value='debts_owed_by_committee'>Debts owed by Committee</option>


                </select>
        </div>
    )
}
