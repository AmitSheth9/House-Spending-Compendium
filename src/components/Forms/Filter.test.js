import { screen, render } from '@testing-library/react'
import Filter from './Filter'

it('should render Filter component', () => {
    const {container} = render(<Filter />);
    const selectDisbursements = screen.getByText(/Disbursements/);
    const selectReceipts = screen.getByText(/Receipts/);
    

    expect(selectDisbursements).toBeInTheDocument();
    expect(selectReceipts).toBeInTheDocument();
    expect(container).toMatchSnapshot();
});