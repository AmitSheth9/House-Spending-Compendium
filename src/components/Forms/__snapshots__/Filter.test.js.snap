// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Filter component 1`] = `
<div>
  <div>
    <select>
      <option
        value="disbursements"
      >
        Disbursements
      </option>
      <option
        value="receipts"
      >
        Receipts
      </option>
      <option
        value="cash_on_hand_end_period"
      >
        Cash on Hand
      </option>
      <option
        value="debts_owed_by_committee"
      >
        Debts owed by Committee
      </option>
    </select>
  </div>
</div>
`;
